sdk/nodejs/runtime/resource.ts

```diff
@@ -114,6 +114,12 @@ export function registerResource(res: Resource, t: string, name: string, custom:
 
     const monitor: any = getMonitor();
     const resopAsync = prepareResource(label, res, custom, props, opts);
+
+    // In order to present a useful stack trace if an error does occur, we preallocate potential
+    // errors here. V8 captures a stack trace at the moment an Error is created and this stack
+    // trace will lead directly to user code. Throwing in `runAsyncResourceOp` results in an Error
+    // with a non-useful stack trace.
+    const preallocError = new Error();
```

Comment:
I preallocated an `Error` here so that we can give users a better stack trace when `RegisterResource` and friends throw exceptions. This `registerResource` will be called on the same "stack" as the user's code that created a resource, so the stack trace will show the user code that led to this resource being created.